compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -659,10 +671,13 @@ OMR::ARM64::TreeEvaluator::vdivEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 TR::Register *
 OMR::ARM64::TreeEvaluator::vandEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
+   TR_ASSERT_FATAL_WITH_NODE(node, node->getDataType().getVectorLength() == TR::VectorLength128,
+                   "Only 128-bit vectors are supported %s", node->getDataType().toString());
+
    TR::InstOpCode::Mnemonic andOp;
-   switch(node->getDataType())
+   switch(node->getDataType().getVectorElementType())
       {
-      case TR::VectorInt8:
+      case TR::Int8:
```

Comment:
The original code does not accept element types other than 8-bit integer, but there is no reason not to accept other integer types. Same for other logical bitwise operations (`vor`, `vxor` and `vnot`).

Reply:
I agree. I will extend those evaluators once this PR gets merged.